can_tools/scrapers/official/TN/tn_state.py

```diff
@@ -0,0 +1,208 @@
+                unit="test_encounters",
+            ),
+            "NEG_TESTS": CMU(
+                category="unspecified_tests_negative",
+                measurement="cumulative",
+                unit="test_encounters",
+            ),
+            "TOTAL_TESTS": CMU(
+                category="unspecified_tests_total",
+                measurement="cumulative",
+                unit="test_encounters",
+            ),
+        }
+
+        # Move things into long format
+        df = df.melt(id_vars=["dt"], value_vars=crename.keys()).dropna()
+
+        # Determine the category of each observation
+        df = self.extract_CMU(df, crename)
+
+        # Determine what columns to keep
+        cols_to_keep = [
+            "dt",
+            "category",
+            "measurement",
+            "unit",
+            "age",
+            "race",
+            "sex",
+            "value",
+        ]
+
+        # Drop extraneous columns
+        out = df.loc[:, cols_to_keep]
+
+        # Convert value columns
+        out["value"] = out["value"].astype(int)
+
+        # Add rows that don't change
+        out["location"] = self.state_fips
+        out["vintage"] = self._retrieve_vintage()
+
+        return out
+
+    def validate(self, df, df_hist) -> bool:
+        "The best developers write test cases first..."
+        # TODO: Create a calculated column and sanity check this against unspecified_tests_total
+        # df["unspecified_tests_total_calculated"] = df.eval("unspecified_tests_positive + unspecified_tests_negative")
+        return True
+
+
+class TennesseeCountyCasesDeathsTests(CMU, CountyDashboard):
+    """
+    Fetch county level covid data from official state of Tennessee spreadsheet
+    """
+
+    source = (
+        "https://www.tn.gov/content/tn/health/cedep/ncov/data.html"
+        "https://www.tn.gov/health/cedep/ncov/data/downloadable-datasets.html"
+    )
+    state_fips = int(us.states.lookup("Tennessee").fips)
+    has_location = False
+    location_type = "county"
+
+    def fetch(self) -> requests.Response:
+        # Set url of downloadable dataset
+        url = (
+            "https://www.tn.gov/content/dam/tn/health/documents/cedep/"
+            "novel-coronavirus/datasets/Public-Dataset-County-New.XLSX"
+        )
+        request = requests.get(url)
+        return request.content
+
+    def normalize(self, data) -> pd.DataFrame:
+        # Read data into data frame
+        df = pd.read_excel(data, parse_dates=["DATE"])
+
+        # Rename columns
+        df = df.rename(columns={"DATE": "dt", "COUNTY": "location_name"})
+
+        # Create dictionary for columns to map
+        crename = {
+            "TOTAL_CONFIRMED": CMU(
+                category="cases", measurement="cumulative", unit="people"
+            ),
+            "TOTAL_PROBABLE": CMU(category="cases", measurement="new", unit="people"),
+            "POS_TESTS": CMU(
+                category="unspecified_tests_positive",
```

Comment:
see comment above about finding type of test

Reply:
Updated code as well here.